### PR TITLE
clib.Session: Wrap the GMT_Get_Common API function

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Wrap the `GMT_Get_Common` API function. 

- API documentation: https://docs.generic-mapping-tools.org/dev/devdocs/api.html#inquiring-about-the-gmt-common-options
- C source code: https://github.com/GenericMappingTools/gmt/blob/35be2deca809ba32c6fef89a0833f6b1f5ddebbc/src/gmt_api.c#L13703

Wrapping this function is a necessary step toward addressing other issues/enhancements:

- [ ] https://github.com/GenericMappingTools/pygmt/issues/2401 and 
- [ ] Let `Figure.tilemap` support syntax like `region="JP"` https://github.com/GenericMappingTools/pygmt/pull/2394#discussion_r1125417383

### CODE REVIEW

File: pygmt/clib/session.py
Comment (seisman) [inline pygmt/clib/session.py:512]:
```suggestion
        value : bool, int, float, or numpy.ndarray
```

File: pygmt/clib/session.py
Comment (seisman) [inline pygmt/clib/session.py:516]:
```suggestion
            the return value depends on the choice of the option.
```

File: pygmt/clib/session.py
Comment (seisman) [inline pygmt/clib/session.py:558]:
```suggestion
            raise GMTInvalidInput(f"Unknown GMT common option flag '{option}'.")
```

File: pygmt/clib/session.py
Comment (seisman) [inline pygmt/clib/session.py:555]:
```suggestion
        pygmt.exceptions.GMTInvalidInput: Unknown GMT common option flag 'A'.
```

**Review by michaelgrund — APPROVED** [review]

**Comment by seisman** [comment]
Hopefully, this PR can get another review before I merge it @GenericMappingTools/pygmt-maintainers

File: pygmt/clib/session.py
Comment (weiji14) [inline pygmt/clib/session.py:527]: Actually, are b and f the same? binary (b) and coltypes (f)?
```suggestion
            - ``"b"``: the option that was set for input or output (int)
            - ``"f"``: the option that was set for input or output (int)
```

Comment (seisman) [reply pygmt/clib/session.py:527]: No, they're not the same.

Initially, I thought the function `GMT_Get_Common` should return the value of the option. For example, after running a command like `gmt plot input.txt -R0/10/0/10 -JX10c/10c -Baf -X2c -Y2c -bi1f -fi0x,1y`, `GMT_Get_Common` should return `X10c/10c` for `J` option and return `i0x,1y` for `f` option, but it doesn't. 

That's because the `GMT_Get_Common` function can only return a status (the function return value) or a double array (function parameter passed by reference). Thus, for options like `-b` and `-f`, the function can only tell if the option was set for input or output (i.e., if `-bi`/`-fi` or `-bo`/`-fo` was specified).

Comment (seisman) [inline pygmt/clib/session.py:527]:
```suggestion
            - ``"b"``: return 0 if `-bi` was set and 1 if `-bo` was set (int)
            - ``"f"``: return 0 if `-fi` was set and 1 if `-fo` was set (int)
```

File: pygmt/clib/session.py
Comment (seisman) [inline pygmt/clib/session.py:533]:
```suggestion
            - ``":"``: return 0 if `-:i` was set and 1 if `-:o` was set (int)
```

File: pygmt/clib/session.py
Comment (weiji14) [inline pygmt/clib/session.py:544]: Does GMT_Get_Common return numbers in inches only? I.e. even if we do `-JX5c/2.5c`, the output value is still converted from centimeters to inches?

File: pygmt/clib/session.py
Comment (weiji14) [reply pygmt/clib/session.py:527]: Ah ok, thanks for the explanation. I think your changes in https://github.com/GenericMappingTools/pygmt/pull/2500#discussion_r1173233018 look better.

File: pygmt/clib/session.py
Comment (seisman) [reply pygmt/clib/session.py:544]: > Does GMT_Get_Common return numbers in inches only? I.e. even if we do `-Xw+1c`, the output value is still converted from centimeters to inches?

Yes, I believe internally GMT stores all dimensions in inches or points, most likely because PostScript uses points as its default unit (https://en.wikipedia.org/wiki/PostScript#Units_of_length), and 72 points are exactly 1 inch, but 72/2.54 points are 1 cm.

Comment (seisman) [reply pygmt/clib/session.py:544]: > > Does GMT_Get_Common return numbers in inches only? I.e. even if we do `-Xw+1c`, the output value is still converted from centimeters to inches?
> 
> Yes, I believe internally GMT stores all dimensions in inches or points, most likely because PostScript uses points as its default unit (https://en.wikipedia.org/wiki/PostScript#Units_of_length), and 72 points are exactly 1 inch, but 72/2.54 points are 1 cm.

That's also why I use `-Xw+1i` in the docstrings, which makes the doctest easier to read.

Comment (seisman) [reply pygmt/clib/session.py:544]: https://github.com/GenericMappingTools/gmt/blob/8f11a6f2ebd665f825e50564e8286433be439f9f/src/gmt_init.c#L1168

This line shows that GMT always converts *xshift* to inches.

File: pygmt/clib/session.py
Comment (weiji14) [inline pygmt/clib/session.py:528]: Looking at https://docs.generic-mapping-tools.org/6.4/gmt.html#h-full, is this just checking whether `-h+d` was set?

File: pygmt/clib/session.py
Comment (weiji14) [inline pygmt/clib/session.py:497]: Please add this function to `doc/api/index.rst`.

File: pygmt/clib/session.py
Comment (weiji14) [inline pygmt/clib/session.py:519]: Should we mention `bool` like this?
```suggestion
            - options ``"B"``, ``"J"``, ``"U"``, ``"g"``, ``"n"``, ``"p"``,
              and ``"s"``: return ``True`` if set, else ``False`` (bool)
```

Comment (seisman) [inline pygmt/clib/session.py:519]:
```suggestion
            - options ``"B"``, ``"J"``, ``"U"``, ``"g"``, ``"n"``, ``"p"``,
              and ``"s"``: return ``True`` (bool)
```

File: pygmt/clib/session.py
Comment (seisman) [reply pygmt/clib/session.py:497]: Done in 9fb950db28755243a3785bd67d464f747579bc34.

File: pygmt/clib/session.py
Comment (seisman) [reply pygmt/clib/session.py:528]: Yes.

**Review by weiji14 — APPROVED** [review]

**Pull request merged**
